proto/catalog.proto

```diff
@@ -405,6 +405,25 @@ message Table {
   // tables and tests. Not to be confused with the global catalog version for
   // notification service.
   TableVersion version = 100;
+
+  // If a materialize view, sink, or table has a node_label, its actors can only be scheduled 
+  // on compute nodes with that label.
+  // If a it does not have a node_label (empty map or null value) it can be scheduled on any 
+  // Compute node without a label. The node has to have both a matching key and value.
+  // E.g. {"some":"value"} can be scheduled on a node with the labels
+  // { 
+  //   "other":"thing",
+  //   "some":"value"
+  // }
+  // but {"some":"thing"} can not be scheduled on that node.
+  // A materialized view or sink or table with an empty node_labels map can only be scheduled on nodes with 
+  // empty node_labels map. 
+  // A node with an empty node_labels map can only host streaming jobs with an empty node_labels map. 
+  map<string, string> node_labels = 101;
```

Comment:
What is the relations of the `node_labels` provided here? All (AND) or any (OR) of them, which one is it?

Reply:
I would suggest the following operation: If `streaming_job.node_labels` is subset or equal to `node.node_labels` then schedule streaming_job on node. 

E.g.
```
node: {
  "a": "b",
  "c": "d"
}
```

can host
```
mv1: {
  "a": "b"
}
mv1: {
  "c": "d"
}
mv3: {
  "a": "b",
  "c": "d"
}
```

can not host
```
mv1: {
  "otherKey": "b"
}
mv1: {
  "c": "otherValue"
}
mv3: {
  "a": "b",
  "c": "d",
  "additional": "value"
}
```

Reply:
I think it's unreasonable to store `node_labels` and `backfill_done` fields directly in the catalog, which is meaningless and FE won't need it. These fields are actually the status of the streaming job, we'd better persist them in `TableFragments` or `streaming_job` system table(for sql backend). I suggest not to change the catalogs here.